### ISSUE TITLE
Prototype: create new GitHub Action which builds this operator into the stackable-experimental image repo

### ISSUE THREAD

**Issue by vsupalov** [stackabletech/kafka-operator#481]
The learnings will be used to apply this action to all operators.

**Comment by vsupalov** [comment]
A few points which are up for discussion:

1) Running tests both on ARM and AMD during this workflow is an item on my personal wish list, but I'm not sure if it's required at this point. If it is, see [0] below. (Thus the question mark of the acceptance criteria point.)

2) What happens if the workflow fails to build an ARM image? Do we still want to continue and push the AMD image solo?

---

[0] Having hosted ARM runners [can take a while](https://github.com/actions/runner-images/issues/2187). Self-hosted GitHub Actions runners are discouraged for non-private repositories.

We found a way to work with qemu and produce both architectures on managed AMD runners. There was a lot of gotchas.

Is a slower performance, or the inability to run tests without emulation enough to consider moving away from building images using GitHub Actions? Self-hosting something similar to ARM runners would be a pain in any case.

**Comment by fhennig** [comment]
@lfrancke @vsupalov Did we make a ticket for multi-arch testing? So we don't forget

**Comment by lfrancke** [comment]
I did not, Vladislav, did you?

**Comment by fhennig** [comment]
I found this ticket, which is linked in the epic and covers the ARM testing aspect. Since we already test for am64, I think this is sufficient: https://github.com/stackabletech/t2/issues/357

**Comment by vsupalov** [comment]
That part felt far-off and fuzzy at the point of planning. It still does.

The new ticket looks reasonable, thanks for adding it! There is a mention of "Run test infrastructure for AMD and ARM images which are pushed to stackable-experimental" in the final tasks, but there's no reason for it to be blocked by everything else.

**Comment by Maleware** [comment]
IIRC this is blocked by the tools image too. Think we should first solve the tools thingy before picking this up.

**Comment by lfrancke** [comment]
Thanks. The tools image stuff is not on the board at all. We need to bring it up at the arch meeting.

**Comment by Maleware** [comment]
It's in the agenda

**Comment by sbernauer** [comment]
Putting in Track column as we are not sure on how to progress with ARM (prioritization wise)

**Comment by sbernauer** [comment]
@lfrancke should we remove it from the board entirely?

**Comment by vsupalov** [comment]
Instead of using GitHub actions, we can use CircleCI which offers native ARM environments. See [this PR](https://github.com/stackabletech/operator-templating/pull/210).

**Comment by vsupalov** [comment]
Retiring this spike, as it is outdated.
